### DISCUSSION TITLE
ETTV connecting to etlded

### DISCUSSION THREAD

**Discussion by mittermichal** [etlegacy/etlegacy#2853] [General]
@ryzyk-krzysiek I was testing out https://github.com/etlegacy/etlegacy/commit/81d59c026b0b65e0ee107622afbe3efb0e031947
It worked (I was watching player on etlded+legacy via ettv server connected to it) when in ETTV homepath I copied original tvgame into legacy folder. 
But if I use new legacy tvgame I get 
```
********************
ERROR: Couldn't find a spawn point

********************
```
Is this supposed to happen?

**Comment by ryzyk-krzysiek** [comment]
I just tried etlded legacy mod as master server and connected ettv as slave with legacy tvgame and it works, what map are you playing on, since that error is about map spawnpoints.

**Comment by mittermichal** [comment]
oasis, I will try something common like supply

**Comment by mittermichal** [comment]
It worked for both supply.pk3 and supply_sp.pk3 (minor spawnpoints added)

**Comment by ryzyk-krzysiek** [comment]
I can confirm oasis crashes even on etlded tv server

**Reply by mittermichal** [reply]
same issue on:
- etl_frostbite_v17.pk3
- etl_adlernest_v4.pk3
- et_operation_b5.pk3

**Comment by mittermichal** [comment]
I will try currently played competitive maps if there is any issue with them later.
Or you can. I am trying to get this into gamestv 😄

**Comment by ryzyk-krzysiek** [comment]
No need to test every single map really, the bug is in `TVG_FindIntermissionPoint` and it's straight forward but I don't like `level.intermission_origin` handling from the beginning so I would rather finally make it in a way I would like.

You should use etlded not ettv anyway, it's not guaranteed that legacy tvgame will work fully with ettv always. It was fun to make it so at first but I see no reason for it to be the case if something stands in the way (I don't wanna be held back by it just like 2.6b helds back ETL). I already have a feature planned that will not work in ettv and although it won't break it it will not work properly at all times. It could be made differently but I want to make it in a specific way that mimics cgame (like the rest of code already) instead of using that `G_ETTV` trap from ettv.

Unless of course people will disagree with me and will want to keep ettv fully compatible with legacy tvgame.

**Comment by mittermichal** [comment]
Only need few active 6v6 maps (sw_goldrush_te, etl_adlernest, etl_frostbite, radar, missile_b3, bremen_b3, decay_sw) to test.

Motivation to use ettv is that I assume its easier to just add needed legacy mod pk3 and tvgame to GTVd distribution than to also add additional logic
```
if game==LEGACY: 
  bin=etlded
  fs_game=legacy
elif game==ETPRO:
 bin=ettv.x86
 fs_game=etpro
...
```
Currently there is no distiction in gamestv between etpro and legacy when creating matches, so that would have to added too I suppose. But all this is just my assumption. Using ettv could atleast give us something working 

> I don't wanna be held back by it just like 2.6b helds back ETL

yes, don't be. Some simplier automation of tv-connecting just 1 edlded to game server that would suffice could be made in 1 day, so this doens't have to be bound by what gamestv system can do.

**Comment by ryzyk-krzysiek** [comment]
You should be able to host both master and tv server on etlded on ETPro too, just use etpro tvgame (in etpro fs_game folder), not legacy tvgame. Granted probably noone ever played etpro mod on etlded but all other mods work so it should work too I guess (and you can definitely load it just that _maybe_ there are some bugs but I kinda doubt).

I don't plan on dropping support of etpro on etlded. I've added code to legacy tvgame in case someone/me wants to add support for etpro instead of using the old etpro tvgame but untill then etpro tvgame should work on etlded just fine.

So I would say you could use etlded for both legacy and etpro on gtv, you just need to have 2 different tvgame's right now.

**Reply by mittermichal** [reply]
So that means etlded+(etpro tvgame) playback of etpro tv demos should work too, right?

**Reply by ryzyk-krzysiek** [reply]
Yes, even legacy tvgame should work, just that game commands are broken (for example scoreboard is not working) so it's better to use etpro one.

**Reply by mittermichal** [reply]
Just tried and yes that works too

**Comment by ryzyk-krzysiek** [comment]
This PR addresses the issue https://github.com/etlegacy/etlegacy/pull/2859

**Comment by mittermichal** [comment]
@ryzyk-krzysiek just letting you know that I sent you email to your commit message email, check spam folder probably
